docs/proposals/default-service-plans.md

```diff
@@ -0,0 +1,544 @@
+  * [Developer](#developer)
+     * [Service Class Discovery](#service-class-discovery)
+     * [Legacy Provisioning](#legacy-provisioning)
+     * [Provisioning with Defaults](#provisioning-with-defaults)
+     * [Provisioning by Service Type](#provisioning-by-service-type)
+     * [Service Instance Management](#service-instance-management)
+  * [Helm Charts](#helm-charts)
+     * [Broker](#broker)
+     * [Author](#author)
+     * [Consumer](#consumer)
+* [Implementation](#implementation)
+  * [Service Class](#service-class)
+  * [Service Plan](#service-plan)
+  * [Service Instance](#service-instance)
+  * [Service Binding](#service-binding)
+
+## Abstract
+
+Let's wholesale rip off how default storage classes for volumes work and apply
+it to Service Catalog! 😎 This allows us to specify a dependency on a type of
+service without requiring upfront knowledge of the underlying broker.
+
+> "My application requires a mysql database."
+
+> "I just want to try out this application using the recommended defaults."
+
+> "I need to encourage everyone to use the cheapest plans possible in our
+> integration environment by making a small database the default."
+
+## Summary
+
+1.  Add default provisioning and binding parameters to Service Classes.
+1.  Add a "Service Type" field to classes and plans, e.g. mysql or redis.
+1.  Add a is-default-plan annotation to Service Plans to flag it as the default
+    plan for a particular service type.
+1.  Service Instances and Bindings fall back to the defaults set on the class  
+    when parameters are not provided.
+1.  Services can be provisioned using only the Service Type, e.g. mysql, and the
+    default plan and parameters are used.
+
+## Motivation
+
+Consider a Helm chart author who is interested in updating their chart to be
+"Service Catalog Ready"\*. Their chart requires a MySQL database and they'd like
+to help their users by setting up the database in their chart, just like the
+upstream Helm charts do today.
+
+So they add a ServiceInstance to their chart, and when they get to the class and
+plan fields in the spec, they hit a wall. The author doesn't know the
+capabilities of the cluster on which their chart will be deployed and aren't in
+a position to provide the class and plan. Neither do they know the secret keys,
+as it's broker specific. The author is forced to add required arguments to their
+chart, passing the buck on figuring out the right class, plan, instance
+parameters and secret keys to the person deploying the chart.
+
+Now consider a user who wants to get up and running quickly, so naturally they
+chose to use a "Service Catalog Ready"* Helm chart. They don't know or care
+about the details, and want to get the application deployed with sane defaults
+so that they can try it out. Before they can even install the application they
+need to:
+
+* Realize that they need to specify a class and plan.
+* Figure out what a broker is and which one is installed on their cluster.
```

Comment:
They still need to do this right?

Reply:
My evil plan is to finish implementing [Minibroker](https://github.com/osbkit/minibroker), and trick everyone into installing it by default (initially as an optional plugin). 😀  Then in most cases, where the user is on a managed cluster (like that provided by Azure, RedHat, Google or Amazon) or on a local minikube cluster, they can rely on having a broker by default.

For those people, when using a chart that understands the changes in this proposal, they could be unaware of the broker, service types, plans, etc and things would "just work".

For people who don't have a broker, they would either continue to use the stock charts (which by default install embedded services). Once they learn about brokers they can: pick one, install it, and then use a "Service Catalog Enabled" chart without needing to specify the exact plan, parameters, etc.